src/search_renderer.cpp

```diff
@@ -162,28 +162,28 @@ kainjow::mustache::data buildPagination(
 std::string SearchRenderer::getHtml()
 {
   // Build the results list
-  kainjow::mustache::data results{kainjow::mustache::data::type::list};
-
+  kainjow::mustache::data results;
+  kainjow::mustache::data items{kainjow::mustache::data::type::list};
   for (auto it = m_srs.begin(); it != m_srs.end(); it++) {
     kainjow::mustache::data result;
+    std::string zim_id(it.getZimId());
     result.set("title", it.getTitle());
-    result.set("url", it.getPath());
+    result.set("absolutePath", urlEncode(protocolPrefix+mp_nameMapper->getNameForId(zim_id)+"/"+it.getPath()));
```

Comment:
At first, this line seemed dubious to me because of `protocolPrefix` being inside the call to `urlEncode()`. While checking how `urlEncode()` would handle the colon character, I discovered to my surprise, that by default, `kiwix::urlEncode()` doesn't encode reserved URL symbols (including `&`, `+`, and `?`). 

https://github.com/kiwix/libkiwix/blob/26eccb5a5fad04e2c44f2bb204491750fb0cf328/src/tools/stringTools.h#L43
https://github.com/kiwix/libkiwix/blob/26eccb5a5fad04e2c44f2bb204491750fb0cf328/src/tools/stringTools.cpp#L215-L230
https://github.com/kiwix/libkiwix/blob/26eccb5a5fad04e2c44f2bb204491750fb0cf328/src/tools/stringTools.cpp#L149-L191

I guess we have to review all call sites of `urlEncode()` and, maybe, its default behaviour, too!

Reply:
Yes, I was surprised also the first time I see it, but it is usefull.
Maybe we should duplicate the function (with `urlEncodeAll` and `urlEncodeSoft` ?) to be explicit about what is done and avoid such surprise in the future.

Reply:
Probably that none of the two options are good:
- We don't want to url encode `/` chars as it is part of the normal path
- As there is no query string here, we want to url encode `&`/`?` chars (potentially present in the article path)

From this page https://www.tutorialspoint.com/html/html_url_encoding.htm, we should probably have a more complete choice on what encode depending of which context.